pkg/v11/resource/chartoperator/create.go

```diff
@@ -102,6 +107,65 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 
 			r.logger.LogCtx(ctx, "level", "debug", "message", "created chart-operator chart")
 		}
+
+		var tenantK8sClient kubernetes.Interface
+		{
+			tenantK8sClient, err = r.getTenantK8sClient(ctx, obj)
+			if tenantcluster.IsTimeout(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
+
+				// A timeout error here means that the cluster-operator certificate
+				// for the current guest cluster was not found. We can't continue
+				// without a Helm client. We will retry during the next execution, when
+				// the certificate might be available.
+				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+				resourcecanceledcontext.SetCanceled(ctx)
+
+				return nil
+			} else if guest.IsAPINotAvailable(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "guest API not available")
+
+				// We should not hammer guest API if it is not available, the guest
+				// cluster might be initializing. We will retry on next reconciliation
+				// loop.
+				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+				resourcecanceledcontext.SetCanceled(ctx)
+
+				return nil
+			} else if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+		{
+			// We wait for the chart-operator deployment to be ready so the
+			// chartconfig CRD is installed. This allows the chartconfig
+			// resource to create CRs in the same reconcilation loop.
+			r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for ready chart-operator deployment")
+
+			o := func() error {
+				err := r.checkDeploymentReady(ctx, tenantK8sClient, chartOperatorNamespace, chartOperatorDeployment, 1)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				return nil
+			}
+
+			// Only wait for 2 minutes. If this takes longer the chartconfig CRs
+			// will be created in the next reconciliation loop.
+			b := backoff.NewConstant(2*time.Minute, 15*time.Second)
```

Comment:
How long does it take? I think 2 minutes blocking is too long. We said when it is like 10 or 30 seconds it is okayish.

Reply:
Ack. I'll do some more testing and see if I can reduce it.